frontend/src/pages/Semesters/Creation/index.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useEffect, useState } from "react";
+import "./styles.css";
+import { useNavigate } from "react-router-dom";
+
```

Comment:
Break down into subcomponents, utilize useEffect for child component refreshes. This is hard to understand and maintain, and better to refactor now than later IMO

Reply:
fixed!